test/passing/tests/layout_annotation_immediate_rewrite.ml.ref

```diff
@@ -0,0 +1,31 @@
+(* Legacy immediate attributes are rewritten into layout annotations *)
+
+type old_imm : immediate
+type old_imm64 : immediate64
+type old_imm : immediate = p
+type old_imm64 : immediate64 = q
+
+(* Comments are not dropped *)
+
+type old_imm : immediate
+
+(* a *)
```

Comment:
It's a bit of a shame that these comments end up attached to `old_imm64` instead of `old_imm` here. Is there a way to fix this?

Reply:
Moving the comments to before the type decl seems to yield much better results:

https://github.com/janestreet/ocamlformat/blob/ea50ab1bb795b04ec8eeb7f4defdd6fb5917edbb/test/passing/tests/layout_annotation_immediate_rewrite.ml.ref#L14-L22

Reply:
Yes, this is better. Might be a bit surprising that the comments are reordered relative to each other sometimes, but I think this will be rare and is acceptable.